core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -109,8 +113,8 @@ public void cleanupTables() {
         !name.startsWith("snap") && Files.getFileExtension(name).equalsIgnoreCase("avro")));
   }
 
-  private TestTables.TestTable create(Schema schema, PartitionSpec spec) {
-    return TestTables.create(tableDir, "test", schema, spec);
+  private TestTables.TestTable create(Schema schema, PartitionSpec spec, Boolean accumulateDeletedFiles) {
+    return TestTables.create(tableDir, "test", schema, spec, accumulateDeletedFiles);
```

Comment:
This class doesn't need to change at all because you can use [`ExpireSnapshots.deleteWith`](https://iceberg.apache.org/javadoc/0.7.0-incubating/org/apache/iceberg/ExpireSnapshots.html#deleteWith-java.util.function.Consumer-) and pass a callback. In each test case, you'd do this to get the list of deletes:

```java
  List<String> deletedFiles = Lists.newArrayList();
  table.expireSnapshots()
      .deleteWith(deletedFiles::add)
      .expireOlderThan(t)
      .commit();
```

Reply:
ohh really. Thanks, nice API!